docs/CODING_STYLE.rst

```diff
@@ -67,32 +67,219 @@ Coding guide
+about what you were thinking when the code was written, and **why** you did things
+that way -- information that cannot be conveyed by the source code alone.
+
+
+Doxygen Comment Specifics
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Doxygen is the first program in a data-flow chain that generates the online LVGL
+API documentation from the files in the LVGL repository.  Doxygen detects comments
+it should pay attention to by leading ``/**``.  It ignores comments that do not
+have exactly two ``*``.
+
+    | ``/** Description of code member AFTER this comment. */``
+    | ``code_member_here {...}`` (e.g. function, struct, etc.)
+
+    ``variable_or_struct_member;  /**< Description of code member BEFORE this comment. */``
+
+Add 2 spaces after Doxygen commands (they start with '@') for improved readability.
+
+Use \`...\` around variable names.  This *can* also be done with type, struct and
+function names when it adds clarity or readability.  Append empty "()" after function
+names. Doxygen generates a hyperlink to the type, struct or function's documentation
+when it exists.  Note:  if you put any text inside the "()", Doxygen will not
+generate a hyperlink in the generated coumentation.
+
+    ``@param  disp   `lv_display_t` object created by lv_x11_window_create()``
+
+Always include a brief description of the code member you are documenting.  For
+documentation that appears *before* the code member, if more detail is needed,
+include a blank line below the brief description and add the detail below it.
+(Doxygen needs the blank line to separate the "brief description" from the
+"additional detail", and it treats it accordingly.  The blank line also improves
+readability in the source code.)  If a struct's member needs more than one line
+to adequately describe it, it is acceptable to place the description before the
+member.  In this case, remember to use ``/**``, not ``/**<``
+
+To document a function's arguments, use the ``@param`` Doxygen command followed by 2
+spaces and the name of the argument.  ``[in]``, ``[out]``, or ``[in,out]`` can be
+appended to ``@param`` to clarify direction when it is important for clarity, like this:
+
+    | ``@param[in]   value   input value``
+    | ``@param[in]   factor  multiplier``
+    | ``@param[out]  result  computed result``
+
+Normally the brief description is simply a noun phrase like "``computed result``"
+and so it neither needs to be capitalized nor does it need a period at the end.
+However, when whole sentences are needed for clarity, capitalize the first letter
+and use appropriate punctuation between sentences for clarity.
+
+Align the beginning of each argument description for improved readability.  Provide
+at least 2 spaces after the longest argument name for visual separation (improves
+readability). If a description of an argument continues on subsequent lines, indent
+the continuation lines by an additional 4 spaces to visually distinguish these lines
```

Comment:
This is more readable to me. Does it work too?
```
 * @param  filter     event code (e.g. `LV_EVENT_CLICKED`) indicating which
 *                    event should be called (`LV_EVENT_ALL` can be used
 *                    to receive all events)
 ```

Reply:
> Does it work too?

Certainly there is no loss to the ease of reading it.  In your example, I am counting 5 spaces after the longest argument.  Would you like me to specify 5 spaces as a minimum?

Reply:
I think we shouldn't control the number of spaces just ask for indenting the description.

Reply:
> I think we shouldn't control the number of spaces just ask for uniformly indenting the description. IMO It's enough to show it in the example.

That makes the most sense to me as well!